environments/arcus/hooks/check_slurm.yml

```diff
@@ -0,0 +1,20 @@
+- name: Run sinfo
+  shell: 'sinfo --noheader --format="%N %P %a %l %D %t" | sort' # using --format ensures we control whitespace: Partition,partition_state,max_jobtime,num_nodes,node_state,node_name
+  register: sinfo
+  changed_when: false
+  until: "'boot' not in sinfo.stdout_lines"
+  retries: 5
+  delay: 10
+- name: Check nodes have expected slurm state
+  assert:
+    that: sinfo.stdout_lines == expected_sinfo
+    fail_msg: |
+      sinfo output not as expected:
+      actual:
+      {{ sinfo.stdout_lines }}
+      expected:
+      {{ expected_sinfo }}
+      <end>
+  vars:
+    expected_sinfo:
+      - "{{ openhpc_cluster_name }}-compute-[0-1] {{ openhpc_slurm_partitions[0].name }}* up 60-00:00:00 2 idle"
```

Comment:
Yes, you'd just have to change the `1` in `[0-1]` to be `{{ groups('openhpc_cluster_name' + '_' +openhpc_slurm_partitions[0].name) | length}} ` and the `60-00:00:00` comes from the default role group/partiition [maxtime](https://github.com/stackhpc/ansible-role-openhpc#slurmconf) which falls through to a cluster default. Or you could ask Slurm for it I guess.

Reply:
Aghh you've just made me find a bug: https://github.com/stackhpc/ansible-role-openhpc/issues/136

Reply:
Ah having read your comment again - no in its current form this can't be added to the slurm playbook itself, although it could be somewhere (e.g. ansible/ci) which e.g. CaaS playbooks could use. Because we don't know when the slurm playbook is run, and there might be running jobs. In which case the format is much harder to parse as slurm splits partitions out to multiple lines by node state.